src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultActions = [];
+  let newState = { ...state };
+
+  for (const action of actions) {
+    const { type } = action;
+    const obj = { ...newState };
```

Comment:
Unnecessary line of code you have already made a copy of the state, so you can use it here